### ISSUE TITLE
idris-1.1.0 fails to compile (DeepSeq.hs)

### ISSUE THREAD

**Issue by mneumann** [idris-lang/Idris-dev#3960]
```
gmake: Leaving directory '/tmp/cabal-tmp-1316/idris-1.1.0/rts'
fatal: Not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
Generating dist/dist-sandbox-af9044e9/build/autogen/Version_idris.hs for release
Building idris-1.1.0...
Preprocessing library idris-1.1.0...
[  1 of 114] Compiling Tools_idris      ( src/Tools_idris.hs, dist/dist-sandbox-af9044e9/build/Tools_idris.o )
[  2 of 114] Compiling IRTS.JavaScript.AST ( src/IRTS/JavaScript/AST.hs, dist/dist-sandbox-af9044e9/build/IRTS/JavaScript/AST.o )
[  3 of 114] Compiling Util.Net         ( src/Util/Net.hs, dist/dist-sandbox-af9044e9/build/Util/Net.o )
[  4 of 114] Compiling Util.ScreenSize  ( src/Util/ScreenSize.hs, dist/dist-sandbox-af9044e9/build/Util/ScreenSize.o )
[  5 of 114] Compiling Version_idris    ( src/Version_idris.hs, dist/dist-sandbox-af9044e9/build/Version_idris.o )
[  6 of 114] Compiling Paths_idris      ( dist/dist-sandbox-af9044e9/build/autogen/Paths_idris.hs, dist/dist-sandbox-af9044e9/build/Paths_idris.o )
[  7 of 114] Compiling IRTS.System      ( src/IRTS/System.hs, dist/dist-sandbox-af9044e9/build/IRTS/System.o )
[  8 of 114] Compiling Idris.Help       ( src/Idris/Help.hs, dist/dist-sandbox-af9044e9/build/Idris/Help.o )
[  9 of 114] Compiling Idris.Colours    ( src/Idris/Colours.hs, dist/dist-sandbox-af9044e9/build/Idris/Colours.o )
[ 10 of 114] Compiling Util.System      ( src/Util/System.hs, dist/dist-sandbox-af9044e9/build/Util/System.o )
[ 11 of 114] Compiling Util.DynamicLinker ( src/Util/DynamicLinker.hs, dist/dist-sandbox-af9044e9/build/Util/DynamicLinker.o )
[ 12 of 114] Compiling Util.Pretty      ( src/Util/Pretty.hs, dist/dist-sandbox-af9044e9/build/Util/Pretty.o )
[ 13 of 114] Compiling Idris.Core.TT    ( src/Idris/Core/TT.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/TT.o )
[ 14 of 114] Compiling Idris.Core.CaseTree ( src/Idris/Core/CaseTree.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/CaseTree.o )
[ 15 of 114] Compiling Idris.Core.Constraints ( src/Idris/Core/Constraints.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/Constraints.o )
[ 16 of 114] Compiling Idris.Core.Evaluate ( src/Idris/Core/Evaluate.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/Evaluate.o )
[ 17 of 114] Compiling Idris.Core.DeepSeq ( src/Idris/Core/DeepSeq.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/DeepSeq.o )
[ 18 of 114] Compiling Idris.Core.Unify ( src/Idris/Core/Unify.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/Unify.o )
[ 19 of 114] Compiling Idris.Core.WHNF  ( src/Idris/Core/WHNF.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/WHNF.o )
[ 20 of 114] Compiling Idris.Core.Typecheck ( src/Idris/Core/Typecheck.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/Typecheck.o )
[ 21 of 114] Compiling Idris.Core.ProofTerm ( src/Idris/Core/ProofTerm.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/ProofTerm.o )
[ 22 of 114] Compiling Idris.Core.ProofState ( src/Idris/Core/ProofState.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/ProofState.o )
[ 23 of 114] Compiling Idris.Core.Elaborate ( src/Idris/Core/Elaborate.hs, dist/dist-sandbox-af9044e9/build/Idris/Core/Elaborate.o )
[ 24 of 114] Compiling IRTS.Lang        ( src/IRTS/Lang.hs, dist/dist-sandbox-af9044e9/build/IRTS/Lang.o )
[ 25 of 114] Compiling Idris.Docstrings ( src/Idris/Docstrings.hs, dist/dist-sandbox-af9044e9/build/Idris/Docstrings.o )
[ 26 of 114] Compiling Idris.Unlit      ( src/Idris/Unlit.hs, dist/dist-sandbox-af9044e9/build/Idris/Unlit.o )
[ 27 of 114] Compiling IRTS.Defunctionalise ( src/IRTS/Defunctionalise.hs, dist/dist-sandbox-af9044e9/build/IRTS/Defunctionalise.o )
[ 28 of 114] Compiling IRTS.Simplified  ( src/IRTS/Simplified.hs, dist/dist-sandbox-af9044e9/build/IRTS/Simplified.o )
[ 29 of 114] Compiling IRTS.CodegenCommon ( src/IRTS/CodegenCommon.hs, dist/dist-sandbox-af9044e9/build/IRTS/CodegenCommon.o )
[ 30 of 114] Compiling Idris.Options    ( src/Idris/Options.hs, dist/dist-sandbox-af9044e9/build/Idris/Options.o )
[ 31 of 114] Compiling Idris.AbsSyntaxTree ( src/Idris/AbsSyntaxTree.hs, dist/dist-sandbox-af9044e9/build/Idris/AbsSyntaxTree.o )
[ 32 of 114] Compiling Idris.DeepSeq    ( src/Idris/DeepSeq.hs, dist/dist-sandbox-af9044e9/build/Idris/DeepSeq.o )

src/Idris/DeepSeq.hs:32:10:
    Duplicate instance declarations:
      instance NFData CT.Options -- Defined at src/Idris/DeepSeq.hs:32:10
      instance NFData CT.Options -- Defined in `Cheapskate.Types'

src/Idris/DeepSeq.hs:35:10:
    Duplicate instance declarations:
      instance NFData CT.ListType
        -- Defined at src/Idris/DeepSeq.hs:35:10
      instance NFData CT.ListType -- Defined in `Cheapskate.Types'

src/Idris/DeepSeq.hs:39:10:
    Duplicate instance declarations:
      instance NFData CT.CodeAttr
        -- Defined at src/Idris/DeepSeq.hs:39:10
      instance NFData CT.CodeAttr -- Defined in `Cheapskate.Types'

src/Idris/DeepSeq.hs:42:10:
    Duplicate instance declarations:
      instance NFData CT.NumWrapper
        -- Defined at src/Idris/DeepSeq.hs:42:10
      instance NFData CT.NumWrapper -- Defined in `Cheapskate.Types'
cabal: Error: some packages failed to install:
idris-1.1.0 failed during the building phase. The exception was:
ExitFailure 1
```

This is using ghc 7.10.2 and cabal 1.22.4.0, on DragonFlyBSD (where ghc 8 is not available yet). I was able to compile Idris with ghc 7.10 before, so I am wondering if it's still possible.

**Comment by melted** [comment]
It seems like cheapskate has added instances that we define today.  So I guess we either need to keep cheapskate back or remove those and raise the lower bound of cheapskate.

**Comment by melted** [comment]
I have lowered the bound for now https://github.com/idris-lang/Idris-dev/commit/9831b017baef3096bae38b44e07da368deb2788e

**Comment by melted** [comment]
@edwinb can you update the metadata for 1.1.0 on hackage to match?

**Comment by mneumann** [comment]
https://github.com/idris-lang/Idris-dev/commit/9831b017baef3096bae38b44e07da368deb2788e fixed it. Thanks!

**Comment by bendiken** [comment]
Attempting to install 1.1.0 with `brew upgrade idris` on macOS also fails due to this error. Would it be possible to push out a 1.1.0.1 release that fixes the installation issues for 1.1.0?

**Comment by melted** [comment]
There is a 1.1.1 on the way.
